napari/_qt/dialogs/qt_notification.py

```diff
@@ -282,9 +275,10 @@ def setup_buttons(self, actions: ActionSequence = ()):
         """
         if isinstance(actions, dict):
             actions = list(actions.items())
+
         for text, callback in actions:
             btn = QPushButton(text)
-            btn.clicked.connect(callback)
+            btn.clicked.connect(lambda: callback(self))
             btn.clicked.connect(self.close)
```

Comment:
I also tried to comment this line to prevent the closing of the toast; to no effects.

Reply:
Yes, the lambda there won't work see https://github.com/napari/napari/pull/1800#issuecomment-749356952 for a detailed explanation with links